homeassistant/components/garage_door/zwave.py

```diff
@@ -25,7 +25,7 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
     node = zwave.NETWORK.nodes[discovery_info[zwave.ATTR_NODE_ID]]
     value = node.values[discovery_info[zwave.ATTR_VALUE_ID]]
 
-    if value.command_class != zwave.COMMAND_CLASS_SWITCH_BINARY or \
+    if value.command_class != zwave.COMMAND_CLASS_SWITCH_BINARY and \
```

Comment:
Hmmm.. This can't be right. If we only have one of the commandclasses this will fail, and initialize .
Shouldn't we want it to only pass along just either, not all? Now it is: If we don't have Binary AND Barrier return otherwise continue.